torch/csrc/api/include/torch/nn/functional/embedding.h

```diff
@@ -1,12 +1,100 @@
 #pragma once
 
+#include <torch/nn/options/embedding.h>
+
 namespace torch {
 namespace nn {
 namespace functional {
 
 inline Tensor one_hot(const Tensor& tensor, int64_t num_classes = -1) {
   return torch::one_hot(tensor, num_classes);
 }
+
+inline Tensor embedding(const Tensor& input, Tensor weight, EmbeddingOptions options = {}) {
+  if (options.padding_idx() != c10::nullopt) {
+    if (*options.padding_idx() > 0) {
+      TORCH_CHECK(*options.padding_idx() < weight.size(0), "Padding_idx must be within num_embeddings");
+    }
+    else if (*options.padding_idx() < 0) {
+      TORCH_CHECK(*options.padding_idx() >= -weight.size(0), "Padding_idx must be within num_embedding");
+      options.padding_idx(weight.size(0) + *options.padding_idx());
+    }
+  } else {
+    options.padding_idx(-1);
+  }
+
+  if (options.max_norm() != c10::nullopt) {
+    torch::NoGradGuard no_grad;
+    torch::embedding_renorm_(weight, input.contiguous(), *options.max_norm(), options.norm_type());
+  }
+  return torch::embedding(weight, input.contiguous(), *options.padding_idx(), options.scale_grad_by_freq(), options.sparse());
+}
+
+inline Tensor embedding_bag(
+    const Tensor& input,
+    Tensor weight,
+    const EmbeddingBagOptions& options = {}) {
```

Comment:
maybe we can pass `options` by value as well, to remove the need for `offsets_` and `per_sample_weights_` :D